### DISCUSSION TITLE
NodeJS - Login with JWT token.

### DISCUSSION THREAD

**Discussion by bijsven** [pocketbase/pocketbase#4529] [Q&A] [answered]
Hello there! I'm currently developing a local app that requires connecting to PocketBase to retrieve data from the user interface, which is hosted on the website. The local app is built on NodeJS, and I aim to store the JWT token to facilitate automatic login if the token remains valid. Is there a method in the JS-sdk that allows for the insertion of the JWT token?

Sorry for the issue that was created :(

**Comment by ganigeorgiev** [comment, marked as answer]
>  The local app is built on NodeJS, and I aim to store the JWT token to facilitate automatic login if the token remains valid. Is there a method in the JS-sdk that allows for the insertion of the JWT token?

I'm not sure that I understand what do you mean by this.

PocketBase is fully stateless and doesn't store the issued tokens anywhere.
If you are asking for a way to load an existing token string into the JS SDK, you can use `pb.authStore.save(token)` (_see all [available authStore fields](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#common-auth-store-fields-and-methods)_) but note that this doesn't persist it as by default the `authStore` uses `LocalStorage` and that is not available in Node and you may need a [custom store](https://github.com/pocketbase/js-sdk#custom-auth-store) if you want to implement your own persistence (eg. loading from a file).

In general if you are trying to use PocketBase as admin from your Node app, you can copy the example admin client from https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130.

**Reply by bijsven** [reply]
I didn't know that pb.authStore.save(token) existed. Now it works perfectly!

Thanks
